src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,234 @@
+import { ModalTitle } from '../ModalTitle/ModalTitle';
+
+type Variant = 'brand' | undefined;
+
+type ModalContentProps = {
+  /**
+   * Optional aria-label for the modal.
+   *
+   * If undefined, the headingText of the Modal.Header will be used.
+   * If there is no Modal.Header, an ariaLabel is required.
+   */
+  ariaLabel?: string;
+  /**
+   * Additional classnames passed in for styling.
+   */
+  className?: string;
+  /**
+   * Contents of the modal.
+   */
+  children: ReactNode;
+  /**
+   * Hides the close button in the top right of the modal.
+   */
+  hideCloseButton?: boolean;
+  /**
+   * A ref to an element that should receive focus when the modal first opens.
+   *
+   * If undefined, the first focusable element (usually the close button) will be used.
+   *
+   * @example
+   * const inputFieldRef = useRef();
+   *
+   * <Modal initialFocus={inputFieldRef}>
+   *   ...
+   *   <InputField ref={inputFieldRef} />
+   * </Modal>
+   */
+  initialFocus?: MutableRefObject<HTMLElement | null>;
+  /**
+   * Toggles scrollable variant of the modal. If modal is scrollable, footer is not, and vice versa.
+   * Defaults to false since modal default is not scrollable.
+   * Also adds border and shadow to the footer indicate sticky status.
+   * Prop should be dependent on whether content overflows at the mobile level.
+   */
+  isScrollable?: boolean;
+  /**
+   * Method called when the close button is clicked. Use this to hide the modal.
+   *
+   * This is required even if you don't have a close button so the ESC key can close the modal.
+   */
+  onClose: () => void;
+  /**
+   * Color variants of the modal.
+   */
+  variant?: Variant;
+};
+
+type ModalProps = ModalContentProps & {
+  /**
+   * Whether or not the modal is visible.
+   */
+  open: boolean;
+};
+
+type Context = {
+  isScrollable?: boolean;
+  variant?: Variant;
+};
+
+const ModalContext = React.createContext<Context>({});
+
+function childrenHaveModalTitle(children?: ReactNode): boolean {
+  const childrenArray = React.Children.toArray(children);
+  return childrenArray.some((child) => {
+    if (typeof child === 'string' || typeof child === 'number') {
+      return false;
+    } else if (
+      'props' in child &&
+      child.type &&
+      typeof child.type !== 'string' &&
+      child.type?.name === 'ModalTitle'
+    ) {
+      return true;
+    } else if ('props' in child && child.props.children) {
+      return childrenHaveModalTitle(child.props.children);
+    }
+    return false;
+  });
+}
+
+/**
+ * The actual modal, without the dark overlay behind it.
+ *
+ * This is only for testing purposes; please do not import and use this directly.
+ */
+export const ModalContent = (props: ModalContentProps) => {
+  const {
+    children,
+    className,
+    hideCloseButton = false,
+    isScrollable,
+    onClose,
+    variant,
+  } = props;
+
+  const componentClassName = clsx(
+    styles['modal__content'],
+    isScrollable && styles['modal__content--scrollable'],
```

Comment:
Actually, should this be the element that has the tab index instead of the body? In the last version of the modal, the header was sticky, so it made sense for the tab index to be one level down.

Reply:
And then also we could only put the tab index on there if the content is scrollable.

Reply:
Oh that would make sense. Let me explore this a bit and also see how the focus indicator looks

Reply:
Having it on `modal__content` causes contrast issue with the brand header and is hidden by the footer

![Screen Shot 2022-06-08 at 4 34 36 PM](https://user-images.githubusercontent.com/86632227/172734774-62b34fc9-64cf-4fe0-afab-b98c4f629904.png)
![Screen Shot 2022-06-08 at 4 35 41 PM](https://user-images.githubusercontent.com/86632227/172734775-519f337c-cc91-49f6-93c3-edbba5ca98be.png)

Reply:
Could you add a comment explaining why the scrollable class is here while the tab index is somewhere else?

Reply:
Yup will do